Updating pg_tde_set_default_key_using_global_key_provider parameter

Andriciuc opened the pull request:
Updates to # Global Principal Key Configuration parameter of setting global key

Andriciuc committed:
Update set-principal-key.md

dAdAbird commented on contrib/pg_tde/documentation/docs/global-key-provider-configuration/set-principal-key.md:
```diff
@@ -4,29 +4,31 @@ You can configure a default principal key using a global key provider. This key
-* `ensure_new_key` defines if a principal key must be unique. The default value `true` means that you must speficy a unique key during key rotation. The `false` value allows reusing an existing principal key.
+* `global_vault_provider` is the name of the key provider you added before. The principal key will be associated with this provider.
+* [optional] third parameter:
+    * `true` forces the creation of a new and unique key.
```
I'd leave only `... a new key`. What uniqueness relates to? A new name implies its uniqueness. If we want to emphasize that the content (actual encryption key) of the new key is secure random generated, we should do it in the fnunction description.

Andriciuc replied:
@dAdAbird I have updated both topics, now they should be more clear/less vague and I was wondering, in the functions chapter should I also address the other parameters containing this 'ensure_new_key'  option? I assume it is similar in function for all parameters, having similar 3 values (empty, true or false), correct? https://internal-pg-tde-docs-pr-351.onrender.com/functions.html#pg_tde_set_key_using_database_key_provider

dAdAbird reviewed (commented):
We also have this: https://pg-tde-docs-pr-351.onrender.com/functions.html#pg_tde_set_key_using_database_key_provider
and think it should be aligned with this PR.

But the description there is a bit confusing.
It says:
```
Creates or rotates the principal key for the current database
```
But those are not mutual exclusive. It might create a new key and rotate existing internal keys with it.

Later it says:
```
The ensure_new_key parameter instructs the function how to handle a principal key during key rotation
```
But it is not exclusively related to rotation. It is applicable to the new keys for database as well.

So in general, the algorithm is next:

```
- If a key with provided name exists:
    - If the 3rd param is true - it fails with the error
    - If the 3rd param is false - it gets this key from the keyring and rotates internal keys with it.
- If a key with provided name does NOT exists:
    -  it creates a new key on the keyring
    -  rotates internal keys for database(s) IF there are any.
```

It's a bit complicated to put all of this together in a comprehensive way since this is a multi-purpose function. But for the user it might by really challenging to get sense of what' going on. 


------

Another thing, on the page subject of this PR, a bit below the proposed changes:
```
!!! note
    The key is auto-generated.
```

Is quite perplexing. What is generated and when? What if key is already existing?

Instead, it might make sense to elaborate a bit on this, possibly in the function description: The user provides only a key name, and we either use it if the key exists or create a new key with the provided name and in this case, the key data (actual encryption key) is auto-generated

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into DOCS-1608-update

Andriciuc committed:
updated functions and set principal key

updated pg_tde_set_key_using_database_key_provider parameter with Andrew's feedback, and rewrote Global Principal Key Configuration to paint a better/more clear picture of key rotation/generation

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into DOCS-1608-update

dAdAbird commented on contrib/pg_tde/documentation/docs/functions.md:
```diff
@@ -264,29 +264,30 @@ Princial keys are stored on key providers by the name specified in this function
-* If set to `true` (default), a new key must be unique.
-  If the provider already stores a key by that name, the function returns an error.
-* If set to `false`, an existing principal key may be reused.
+* `true` (default if omitted): Requires the key to be newly created. If a key with the same name already exists, the function fails.
```
The default value if `false`

Andriciuc replied:
Fixed!

dAdAbird reviewed (changes requested):
LGTM, other than the comment

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into DOCS-1608-update

Andriciuc committed:
updated pg_tde_set_key_using_database_key_provider

set false value as default instead of true as default

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into DOCS-1608-update

nastena1606 commented on contrib/pg_tde/documentation/docs/global-key-provider-configuration/set-principal-key.md:
```diff
@@ -1,40 +1,46 @@
-* `name-of-the-key` is the name of the principal key. You will use this name to identify the key.
-* `provider-name` is the name of the key provider you added before. The principal key will be associated with this provider.
-* `ensure_new_key` defines if a principal key must be unique. The default value `true` means that you must speficy a unique key during key rotation. The `false` value allows reusing an existing principal key.
+* `key-name` the name under which the principal key is stored in the provider.
```
```suggestion
* `key-name` is the name under which the principal key is stored in the provider.
```

nastena1606 commented on contrib/pg_tde/documentation/docs/global-key-provider-configuration/set-principal-key.md:
```diff
@@ -1,40 +1,46 @@
-* `provider-name` is the name of the key provider you added before. The principal key will be associated with this provider.
-* `ensure_new_key` defines if a principal key must be unique. The default value `true` means that you must speficy a unique key during key rotation. The `false` value allows reusing an existing principal key.
+* `key-name` the name under which the principal key is stored in the provider.
+* `global_vault_provider` the name of the global key provider you previously configured.
```
```suggestion
* `global_vault_provider` is the name of the global key provider you previously configured.
```

Andriciuc committed:
Update contrib/pg_tde/documentation/docs/global-key-provider-configuration/set-principal-key.md

updated with fix

Co-authored-by: Anastasia Alexandrova <anastasia.alexandrova@percona.com>

Andriciuc committed:
Update contrib/pg_tde/documentation/docs/global-key-provider-configuration/set-principal-key.md

updated with fix

Co-authored-by: Anastasia Alexandrova <anastasia.alexandrova@percona.com>

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into DOCS-1608-update
